internal/bminventory/inventory.go

```diff
@@ -600,6 +604,19 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
 
+	if b.ocmClient != nil {
+		sub, err := b.ocmClient.AccountsMgmt.CreateSubscription(ctx, id)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to create AMS subscription for cluster %v", id)
+			return common.NewApiError(http.StatusInternalServerError, err)
+		}
+		dbReply := b.db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).Updates(map[string]interface{}{"ams_subscription_id": sub.ID()})
```

Comment:
Please move this code into a function of cluster api - we should not change cluster properties randomly in the code.

Reply:
done.